src/int128.h

```diff
@@ -0,0 +1,182 @@
+#ifndef SECP256K1_INT128_H
+#define SECP256K1_INT128_H
+
+#include <stdint.h>
+#include "util.h"
+
+/*
+#ifdef UINT128_MAX
+#undef UINT128_MAX
+*/
+
+#if defined(UINT128_MAX)
+typedef uint128_t secp256k1_uint128;
+typedef int128_t secp256k1_int128;
```

Comment:
I think it's a idea to define our our own type with a separate prefix. I guess then the entire logic currently in `util.h` will conceptually belong here.

Reply:
I am inclined to move the int128 logic from util.h into this file.  I just haven't done it yet.

Reply:
This hasn't been resolved yet.

Reply:
Are you not happy with the changes to `util.h`?

Reply:
I'd prefer to move the remaining `#ifdef` `WIDEMUL` logic in `util.h` to `int128_impl.h` (or `int128.h`?) because it belongs there conceptually.

I think `util.h` is just the kitchen sink for code pieces that don't have a proper home (or where we were too lazy to create a proper home. Now that you introduce `int128_impl.h` here, I think that's the place where those `#ifdefs` belong.

Reply:
As it stands when SECP256K1_WIDEMUL_INT64 ends up being defined then `int128_impl.h` isn't even imported anywhere.  So I'm not sure what you are imagining.

Reply:
Yes, I was wrong. See below though.